### PR TITLE
Small fixes + 1 new multiplayer level + 2 new skirmish levels

### PR DESCRIPTION
**Pull request by hwoarangmy**

### CODE REVIEW

**Comment by Bertram25** [comment]
What! A new level?

File: source/entities/TreasuryObject.cpp [hunk 1]
```diff
@@ -97,11 +97,18 @@ void TreasuryObject::doUpkeep()
...
+        {
+            removeFromGameMap();
+            deleteYourself();
+            return;
+        }
 
+        // If the gold value changed the mesh, we remove the old mesh and create a new one
         if(getMeshName().compare(getMeshNameForGold(mGoldValue)) != 0)
         {
             // The treasury fullnes changed. We remove the object and create a new one
```
Comment (Bertram25) [inline source/entities/TreasuryObject.cpp:114]: Looks SF.

File: source/modes/MenuModeConfigureSeats.cpp [hunk 2]
```diff
@@ -298,8 +298,8 @@ bool MenuModeConfigureSeats::comboChanged(const CEGUI::EventArgs& ea)
     CEGUI::ListboxItem* selItem = comboSel->getSelectedItem();
     if((selItem != nullptr) &&
        (comboSel->getName().compare(0, COMBOBOX_PLAYER_PREFIX.length(), COMBOBOX_PLAYER_PREFIX) == 0) &&
-       (selItem->getID() != 0) && // Can be several inactive players
-       (selItem->getID() != 1)) // Can be several AI players
+       (static_cast<int32_t>(selItem->getID()) >= Seat::PLAYER_ID_HUMAN_MIN) && // Can be several AI players
+       (static_cast<int32_t>(selItem->getID()) != Seat::PLAYER_TYPE_INACTIVE_ID)) // Can be several inactive players
```
Comment (Bertram25) [inline source/modes/MenuModeConfigureSeats.cpp:302]: Good thing to replace magic numbers :)

File: config/spells.cfg [hunk 3]
```diff
@@ -49,8 +49,8 @@
     CallToWarNbTurnsMax	50
     CallToWarCooldown	10
     CreatureExplosionPrice	5000
-    CreatureExplosionDuration	5
-    CreatureExplosionValue	5
+    CreatureExplosionDuration	3
+    CreatureExplosionValue	14
     CreatureExplosionCooldown	16
     CreatureHastePrice	2000
     CreatureHasteDuration	10
```
Comment (Bertram25) [inline config/spells.cfg:56]: Looks SF. Will find some victims to try that. >:)

File: source/modes/MenuModeConfigureSeats.cpp (hunk 2 above)
Comment (hwoarangmy) [inline source/modes/MenuModeConfigureSeats.cpp:302]: Yes. It would have avoided to forget as I did ^^

**Comment by hwoarangmy** [comment]
> What! A new level?

Yes ^^
Note that I've created a small java program that turns an image to a OD level. If you look closely, you should see our old background mirrored 4 times ^^

**Comment by Bertram25** [comment]
> Note that I've created a small java program that turns an image to a OD level. If you look closely, you should see our old background mirrored 4 times ^^

Cool :) Maybe we should integrate that somehow. :)

**Comment by Bertram25** [comment]
Can be merged, as it is working fine as for me.

**Comment by hwoarangmy** [comment]
> Cool :) Maybe we should integrate that somehow. :)

Why not ? But in this case, I would need to make some basic GUI as everything is hardcoded for now. I just did a quick and dirty program to do the task. Java has native ways of opening image files and it took 15 minutes to do the program. To open the file:

```
BufferedImage image = ImageIO.read(new File("c:/temp/file.png"));
```

Then, to get a pixel value:

```
int pixel = image.getRGB(x, y);
```

That's almost it ^^

**Comment by Bertram25** [comment]
> Java has native ways of opening image files and it took 15 minutes to do the program.

Eh eh. Well, java has a native way for many things, I'm not surprised ;)

> Why not ? But in this case, I would need to make some basic GUI as everything is hardcoded for now.

Ok, let's keep that in mind. Actually, this feature was already started quite some time ago, but never finished, that's why it was removed.

**Comment by Bertram25** [comment]
I have to play tonight. :)

**Comment by Bertram25** [comment]
Working fine. I'll merge.

**Comment by akien-mga** [comment]
Lol I totally missed that we had a new level xD I really to find some time to play the current branch a bit :)

**Comment by akien-mga** [comment]
> I totally missed that we had a new level xD

_Three_ new levels, wow :D

**Pull request merged**